internal/infrastructure/elastic_search/query_device_features.go

```diff
@@ -0,0 +1,141 @@
+package elasticsearch
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+type DeviceFeatures struct {
+	BatteryVoltage struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	FuelPercentRemaining struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Odometer struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Oil struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Soc struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Speed struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Tires struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+}
+
+type DeviceFeaturesResp struct {
+	Aggregations struct {
+		Features struct {
+			SumOtherDocCount int64 `json:"sum_other_doc_count"`
+			Buckets          []struct {
+				Key               string `json:"key"`
+				DocumentCount     int64  `json:"doc_count"`
+				DeviceDefinitions struct {
+					SumOtherDocCount int64 `json:"sum_other_doc_count"`
+					Buckets          []struct {
+						Key           string `json:"key"`
+						DocumentCount int64  `json:"doc_count"`
+						Features      struct {
+							Buckets map[string]map[string]int
```

Comment:
It might be worth it to have this be
```suggestion
							Buckets map[string]struct{
								DocCount int `json:"doc_count"`
							}
```

Reply:
The structure of the response is like this 

``` json
"buckets": {
  "batteryVoltage": {
	  "doc_count": 0
  },
  "fuelPercentRemaining": {
	  "doc_count": 0
  },
  "odometer": {
	  "doc_count": 0
  },
...
```

Reply:
Are we disagreeing? Maybe I'm too sleepy.